plutus-core/untyped-plutus-core/test/DeBruijn/FlatNatWord.hs

```diff
@@ -0,0 +1,51 @@
+module DeBruijn.FlatNatWord (test_flatNatWord) where
+
+import Common
+import Flat
+import Hedgehog
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import Numeric.Natural
+import Test.Tasty.Hedgehog
+
+{- NOTE: [DeBruijjn index flat format]
+In the debruijn branch we are transitioning the debruijn index to be a Word instead of Natural, for CEK performance reasons.
+
```

Comment:
I see that Haskell doesn't specify the exact size of `Word`. The documentation says

>   A [Word](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Word.html#t:Word) is an unsigned integral type, with the same size as [Int](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Int.html#t:Int).

and `Data.Int` says

> A fixed-precision integer type with at least the range [-2^29 .. 2^29-1]. The exact range for a given implementation can be determined by using [minBound](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:minBound) and [maxBound](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:maxBound) from the [Bounded](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:Bounded) class.

Is that a possible danger in future wrt serialisation in the (presumably very unlikely) event that the implementation of `Word` changes?  Would `Word64` or something be safer?

Reply:
Good point, it's also architecture dependent. Let's use a fixed-size word.

Reply:
indeed, to be on the safe size we should change all machine-dependent types (int,word,etc) to fixed size (int64,word64,etc)

Reply:
It depends! In some cases it's fine, e.g. it's fine to use it for internal indices in, say, an environment. But for anything that's externally facing or appears in a format or something then yeah, we should use fixed-size types.